### PR TITLE
Add sample for bentoml

### PR DESCRIPTION
**Pull request by yubozhao**
**What this PR does / why we need it**:

Add code sample for deploying InferenceServer with BentoML

### CODE REVIEW

File: docs/samples/bentoml/README.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+# Predict on an InferenceService using BentoML
+
+This example builds an iris classification model API server with BentoML and then deploys
+it as an InferenceService to a KFServing installed cluster.
+
+[BentoML](https://bentoml.org) is an open-source framework for high-performance ML model
```
Comment (parano) [inline docs/samples/bentoml/README.md]: BentoML is an open-source platform for high-performance ML model serving. It makes building production API endpoint for your ML model easy and support all major machine learning training frameworks, including Tensorflow, Keras, PyTorch, XGBoost, scikit-learn etc.

BentoML comes with a high-performance API model server with adaptive micro-batching support, which achieves the advantage of batch processing in online serving. It also provide model management and model deployment functionality, provides an end-to-end model serving workflow, with DevOps best practices baked in.

File: docs/samples/bentoml/README.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+* Python 3.6 or above
+  * Install required packages `bentoml` and `scikit-learn` on your local system:
+
+    ```shell
+    pip install bentoml scikit-learn
+    ```
+
+### Build API model server using BentoML
+
+The followwing code defines an API model server that requires a `scikit-learn` model, and
```
Comment (parano) [inline docs/samples/bentoml/README.md]: following

File: docs/samples/bentoml/README.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    iris_classifier_service = IrisClassifier()
+
+    # Pack the newly trained model artifact
+    iris_classifier_service.pack('model', clf)
+
+    # Save the prediction service to disk for model serving
+    saved_path = iris_classifier_service.save()
+```
+
+Save and run above code or run the following command to create the iris classifier
```
Comment (parano) [inline docs/samples/bentoml/README.md]: The sample code above can be found in the BentoML repository, run the code with the following command:

```bash
git clone git@github.com:bentoml/BentoML.git
python ./bentoml/guides/quick-start/main.py
```

File: docs/samples/bentoml/README.md (hunk 2 above)
Comment (parano) [reply docs/samples/bentoml/README.md]: ... defines a BentoML prediction service ...

File: docs/samples/bentoml/README.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+@env(auto_pip_dependencies=True)
+@artifacts([SklearnModelArtifact('model')])
+class IrisClassifier(BentoService):
+
+    @api(DataframeHandler)
+    def predict(self, df):
+        return self.artifacts.model.predict(df)
+```
+
+The following code to create a BentoService SavedBundle with the trained an iris
```
Comment (parano) [inline docs/samples/bentoml/README.md]: The following code trains a classifier model and serve it with the `IrisClassifier` defined above:

```python
# main.py
from sklearn import svm
from sklearn import datasets
from iris_classifier import IrisClassifier
if __name__ == "__main__":
    # Load training data
    iris = datasets.load_iris()
    X, y = iris.data, iris.target
    # Model Training
    clf = svm.SVC(gamma='scale')
    clf.fit(X, y)
    # Create a iris classifier service instance
    iris_classifier_service = IrisClassifier()
    # Pack the newly trained model artifact
    iris_classifier_service.pack('model', clf)
    # Save the prediction service to disk for model serving
    saved_path = iris_classifier_service.save()
```
The sample code above can be found in the BentoML repository, run them directly with the
following command:

```bash
git clone git@github.com:bentoml/BentoML.git
python ./bentoml/guides/quick-start/main.py
```

After saving the BentoService instance, you can now start a REST API server with the model trained and test the API server locally:

```bash
# Start BentoML API server:
bentoml serve IrisClassifier:latest
```
```bash
# Send test request:
curl -i \
  --header "Content-Type: application/json" \
  --request POST \
  --data '[[5.1, 3.5, 1.4, 0.2]]' \
  http://localhost:5000/predict
```

File: docs/samples/bentoml/README.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+### Deploy InferenceService
+
+Find the file directory of the SavedBundle with `bentoml get` command, which is
+directory structured as a docker build context. Running docker build with this
+directory produces a docker image containing the API model server. Replace
+`docker_username` with your Docker Hub username and run the following code:
+
+```shell
+# Replace DOCKER_USERNAME with the Docker Hub username
```
Comment (parano) [inline docs/samples/bentoml/README.md]: remove docker_username

File: docs/samples/bentoml/README.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+The sample code above can be found in the BentoML repository, run the code with the
+following command:
+
+```bash
+git clone git@github.com:bentoml/BentoML.git
+python ./bentoml/guides/quick-start/main.py
+```
+
+### Deploy InferenceService
+
```
Comment (parano) [inline docs/samples/bentoml/README.md:108]: BentoML provides a convenient way of containerizing the model API server with Docker.  To create a docker container image for the sample model above:

Find the file directory ...
